httplib.h

```diff
@@ -2665,10 +2665,14 @@ inline std::string if2ip(int address_family, const std::string &ifn) {
         }
       } else if (ifa->ifa_addr->sa_family == AF_INET6) {
         auto sa = reinterpret_cast<struct sockaddr_in6 *>(ifa->ifa_addr);
-        char buf[INET6_ADDRSTRLEN];
-        if (inet_ntop(AF_INET6, &sa->sin6_addr, buf, INET6_ADDRSTRLEN)) {
-          freeifaddrs(ifap);
-          return std::string(buf, INET6_ADDRSTRLEN);
+        auto is_global_addr = ((((const uint32_t *) (&sa->sin6_addr))[0]
+                                & htonl(0x70000000)) == htonl (0x20000000));
```

Comment:
@CyberKoo, could you explain what `is_global_addr` tries to do?

Reply:
this was taken from
https://github.com/pupnp/pupnp/blob/c61460be44645f5c9040c1cf4cf7ac5efbe6b01d/upnp/src/api/upnpapi.cpp#L96
to exclude all reserved IPv6 addresses, more detail
https://en.wikipedia.org/wiki/Reserved_IP_addresses#IPv6
if return local link address (fe80::/64) will cause connection error, since the local link address is only designed for used within LAN.

Reply:
ULA is normally used within a private network, which means this address is normally not accessible from internet (not routable). 
In some special case (NAT66 used), the ULA address will be translated on the router site, so the address become useable.
How about we return ULA if only if there is GUA, otherwise return GUA. 
Does that sound good to you?

Reply:
Do you mean "How about we return ULA if only if there is **not** GUA, otherwise return GUA."?

Reply:
yes

Reply:
Your suggestion sounds good to me.